PWGJE/Tasks/gammajettreeproducer.cxx

```diff
@@ -0,0 +1,328 @@
+
+    double dPhi = TVector2::Phi_0_2pi(phi1) - TVector2::Phi_0_2pi(phi2);
+    if (abs(dPhi) > M_PI) {
+      dPhi = 2 * M_PI - abs(dPhi);
+    }
+    double dEta = eta1 - eta2;
+    return std::sqrt(dPhi * dPhi + dEta * dEta);
+  }
+
+  bool isTrackSelected(const auto& track)
+  {
+    if (!jetderiveddatautilities::selectTrack(track, trackSelection)) {
+      return false;
+    }
+    if (track.pt() < trackMinPt) {
+      return false;
+    }
+
+    return true;
+  }
+
+  bool isEventAccepted(const auto& collision, const auto& bc)
+  {
+    // get bc from collision
+    initCCDB(bc);
+
+    if (mSoftwareTrigger->length() > 0) {
+      if (!zorro.isSelected(bc.globalBC())) {
+        return false;
+      }
+    }
+
+    if (collision.posZ() > mVertexCut) {
+      return false;
+    }
+    if (!jetderiveddatautilities::selectCollision(collision, eventSelection)) {
+      return false;
+    }
+    if (!jetderiveddatautilities::eventEMCAL(collision)) {
+      return false;
+    }
+    return true;
+  }
+
+  double ch_iso_in_cone(const auto& cluster, JetTracks const& tracks, float radius = 0.4)
+  {
+    double iso = 0;
+    for (auto track : tracks) {
+      if (!isTrackSelected(track)) {
+        continue;
+      }
+      // make dR function live somwhere else
+      double dR = deltaR(cluster.eta(), cluster.phi(), track.eta(), track.phi());
+      if (dR < radius) {
+        iso += track.pt();
+      }
+    }
+    return iso;
+  }
+  double ch_perp_cone_rho(const auto& cluster, JetTracks const& tracks, float radius = 0.4)
+  {
+    double ptSumLeft = 0;
+    double ptSumRight = 0;
+
+    double cPhi = TVector2::Phi_0_2pi(cluster.phi());
+
+    // rotate cone left by 90 degrees
+    float cPhiLeft = cPhi - TMath::Pi() / 2;
+    float cPhiRight = cPhi + TMath::Pi() / 2;
+
+    // loop over tracks
+    float dRLeft, dRRight;
+    for (auto track : tracks) {
+      if (!isTrackSelected(track)) {
+        continue;
+      }
+      dRLeft = deltaR(cluster.eta(), cPhiLeft, track.eta(), track.phi());
+      dRRight = deltaR(cluster.eta(), cPhiRight, track.eta(), track.phi());
+
+      if (dRLeft < radius) {
+        ptSumLeft += track.pt();
+      }
+      if (dRRight < radius) {
+        ptSumRight += track.pt();
+      }
+    }
+
+    float rho = (ptSumLeft + ptSumRight) / (2 * TMath::Pi() * radius * radius);
+    return rho;
+  }
+
+  // ---------------------
+  // Processing functions
+  // ---------------------
+
+  // define cluster filter. It selects only those clusters which are of the type
+  // sadly passing of the string at runtime is not possible for technical region so cluster definition is
+  // an integer instead
+  Filter clusterDefinitionSelection = (o2::aod::jcluster::definition == mClusterDefinition);
+  // Process clusters
+  void processClusters(soa::Join<JetCollisions, aod::BkgChargedRhos, aod::JCollisionBCs>::iterator const& collision, aod::JBCs const&, selectedClusters const& clusters, JetTracks const& tracks)
```

Comment:
you can remove JBCs once you remove zorro

Reply:
removed JBCs